src/main.rs

```diff
@@ -148,47 +149,68 @@ async fn main() -> miette::Result<()> {
                 PackageType::Impl
             };
 
-            command::init(kind, package)
+            command::init(kind, package.to_owned())
                 .await
-                .wrap_err(miette!("init command failed"))
+                .wrap_err(miette!(
+                    "failed to initialize {}{kind} package",
+                    package.map(|p| format!("`{p}` as ")).unwrap_or_default()
+                ))
         }
         Command::Add {
             registry,
             dependency,
-        } => command::add(registry, &dependency)
+        } => command::add(registry.to_owned(), &dependency)
             .await
-            .wrap_err(miette!("add command failed")),
-        Command::Remove { package } => command::remove(package)
-            .await
-            .wrap_err(miette!("remove command failed")),
+            .wrap_err(miette!(
+                "failed to add `{dependency}` from `{registry}` to `{MANIFEST_FILE}`"
+            )),
+        Command::Remove { package } => command::remove(package.to_owned()).await.wrap_err(miette!(
+            "failed to remove `{package}` from `{MANIFEST_FILE}`"
+        )),
         Command::Package {
             output_directory,
             dry_run,
         } => command::package(output_directory, dry_run)
             .await
-            .wrap_err(miette!("package command failed")),
+            .wrap_err(miette!(
+                "failed to export `{}` into the buffrs package format",
+                Manifest::read().await?.package.name
```

Comment:
I'm not sure using a fallible function inside of `miette!` is a good idea, have you tested what happens if this fails?